### PR TITLE
Igor: enhanced vlan checking

### PR DESCRIPTION
**Pull request by msymonds**

### CODE REVIEW

File: src/igor/sync.go [hunk 1]
```diff
@@ -82,14 +78,38 @@ func runSync(cmd *Command, args []string) {
...
+	if err != nil {
+		log.Fatal("Unable to acquire VLAN ground truth from arista")
+	}
 	// TODO: probably shouldn't iteration over .M directly
 	for _, r := range igor.Reservations.M {
 		if !r.IsActive(igor.Now) {
 			continue
 		}
 
+		w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', 0)
```
Comment (csymonds) [inline src/igor/sync.go:92]: We should not be creating a new writer in every iteration of this loop. Please move out of the loop.

File: src/igor/sync.go [hunk 2]
```diff
@@ -82,14 +78,38 @@ func runSync(cmd *Command, args []string) {
...
 		if !quiet {
-			fmt.Printf("set switchports for %v to %v\n", r.Hosts, r.Vlan)
+			// print all nodes, Igor VLANs, and arista VLANs
+			fmt.Fprintln(w, "NODE\tIGOR VLAN\tARISTA VLAN")
+			for _, host := range r.Hosts {
```
Comment (csymonds) [inline src/igor/sync.go:96]: You are asking to iterate over single element of an array within the loop of that array. This is not going to have the desired effect.

File: src/igor/sync.go [hunk 3]
```diff
@@ -82,14 +78,38 @@ func runSync(cmd *Command, args []string) {
...
+				vlan := strconv.Itoa(r.Vlan)
+				gtvlan := gt[host]
+				if gtvlan != vlan {
+					vlan = FgRed + vlan + Reset
+				}
+				fmt.Fprintf(w, "%v\t%v\t%v\n", host, vlan, gtvlan)
+			}
+		} else {
+			// just print what's different
+			for _, host := range r.Hosts {
```
Comment (csymonds) [inline src/igor/sync.go:106]: Same as above, you are asking to iterate over single element of an array within the loop of that array. This is not going to have the desired effect

Comment (msymonds) [reply src/igor/sync.go]: Sync iterates over reservations. For each reservation, there is 1 vlan value, but potentially multiple hosts. this inner loop is iterating over the hosts. Each is used to compare the vlan value to the corresponding value in arista using the host as the key.

File: src/igor/sync.go (hunk 1 above)
Comment (msymonds) [reply src/igor/sync.go]: tabwriter moved to outside the loop

File: src/igor/sync.go [hunk 4]
```diff
@@ -82,14 +76,33 @@ func runSync(cmd *Command, args []string) {
...
 	for _, r := range igor.Reservations.M {
 		if !r.IsActive(igor.Now) {
 			continue
 		}
 
 		if !quiet {
+			// print all nodes, Igor VLANs, and arista VLANs
+			for _, host := range r.Hosts {
+				vlan := strconv.Itoa(r.Vlan)
+				fmt.Printf("NODE %v - IGOR VLAN: %v    ARISTA VLAN: %v\n", host, vlan, gt[host])
```
[the remark this answers is not in the document]
Comment (msymonds) [reply src/igor/sync.go]: Report update to this format.

File: src/igor/sync.go [hunk 5]
```diff
@@ -82,14 +76,33 @@ func runSync(cmd *Command, args []string) {
...
+			}
+			// TODO: do we still need this?
 			fmt.Printf("set switchports for %v to %v\n", r.Hosts, r.Vlan)
```
[the remark this answers is not in the document]
Comment (msymonds) [reply src/igor/sync.go]: removed

**Comment by aherna** [comment]
@msymonds @afbagwell have changes been made? Need to close out by next week

**Comment by afbagwell** [comment]
No further work will be done on Igor v1.

**Pull request closed without merging**